App Size: Remove dependency to two larger JS libs (moment and parse5)

nkuehn opened the pull request:
Following our discussion concerning the size and performance,  this PR does 

 * tweak documentation and start scripts on how to do the bundle size analysis. you can run `yarn build:analyze` as a shortcut to run the bundle analysis on 
 * remove the dependency on the full HTML parser "parse5" by doing the metaproperty parsing of code blocks with an own regex based implementation.  (I know, I voted for using the parser, but that was at a time when we needed it for other purposes anyways)
 * remove dependency on moment.js  (time library).  It's only gone from our code now but still in some bundle because it's used in the UI Kit calendar widget.   But it's not in all bundles any more now (it was due the RSS feature) but only in the release notes list.

davifantasia commented on README.md:
````diff
@@ -37,12 +37,15 @@ Build the packages
 ```
 
-Start the [docs-smoke-test website](./websites/docs-smoke-test):
+Start the [docs-smoke-test website](./websites/docs-smoke-test) or the [api-docs-smoke-test website](./websites/api-docs-smoke-test):
 
 ```bash
+$ cd websites/docs-smoke-test
 $ yarn start
 ```
````
Start the [docs-smoke-test website](./websites/docs-smoke-test):

```bash
$ cd websites/docs-smoke-test
$ yarn start
```

Start the [api-docs-smoke-test website](./websites/api-docs-smoke-test):

```bash
$ cd websites/api-docs-smoke-test
$ yarn start
```

nkuehn replied:
much better, right.

davifantasia commented on packages/gatsby-theme-docs/src/layouts/internals/layout-page-navigation.js:
```diff
@@ -174,6 +177,7 @@ const LayoutPageNavigation = (props) => {
 
   const navigationContainer = (
     <nav
+      key="search-container"
```
`key="search-container"` is used twice, here and on line 160

```suggestion
      key="navigation-container"
```

nkuehn replied:
copypasta...  done

davifantasia commented:
I meant to `request changes`, not approve. I tried to change it but it's not working on GitHub at the moment.

nkuehn commented:
> I meant to `request changes`, not `approve`. I tried to change it but it's not working on GitHub at the moment.

No problem, I dismissed and re-requested it. Thanks for the review!

timonrey reviewed (commented):
I tried to run the `create-docs-release-note` script and I got an error here:
<img width="1900" alt="Bildschirmfoto 2021-06-16 um 11 42 11" src="https://user-images.githubusercontent.com/31767369/122196855-e5758580-ce97-11eb-89bb-036376860ae1.png">

Looks like there there is a problem with babel here? Maybe we have different configurations. We need to keep this simple so that the tech-writers and other users can use the script easily.

nkuehn commented:
@timonrey  sorry for not testing properly enough on the release notes script.  I switched the script to directly loading the implementation file and changed the file to CommonJS syntax so it can be run natively in node.js too. 

https://github.com/commercetools/commercetools-docs-kit/pull/986/commits/78325c807ce504c1b2743c67e6524d668eb3e26d

timonrey reviewed (approved):
Thanks 👍🏼
